pkgs/desktops/gnome-3/3.20/core/gnome-shell/default.nix

```diff
@@ -15,7 +16,8 @@ stdenv.mkDerivation rec {
 
   buildInputs = with gnome3;
     [ gsettings_desktop_schemas gnome_keyring gnome-menus glib gcr json_glib accountsservice
-      libcroco intltool libsecret pkgconfig python3 libsoup polkit libcanberra gdk_pixbuf librsvg
+      libcroco intltool libsecret pkgconfig python3 python3Packages.pygobject3
+      libsoup polkit libcanberra gdk_pixbuf librsvg
       clutter networkmanager libstartup_notification telepathy_glib docbook_xsl docbook_xsl_ns
```

Comment:
as I wrote on IRC (but I guess you left by then) I think it would be good if we would use `python.buildEnv` or `python.withPackages` instead. This results in a Python environment where the interpreter can directly find all the Python packages that you need. 

So here you would write
```
(python3Packages.python.withPackages ( ps : with ps; [ pygobject ] ) )
```
instead of `[ python3 python3Packages.pygobject ]`